### DISCUSSION TITLE
Detached Condition from Child in the OT Role

### DISCUSSION THREAD

**Discussion by eric-bj-kang** [openthread/openthread#7864] [Q&A] [answered]
Hello,

I know there are roles in the Openthread like as below.
- Disable / Detached / Child / Router / Leader

And I could find the child detache state in the below link.
- Detached and Leader #7716
  . https://github.com/openthread/openthread/discussions/7716
  . Whenever Thread devices lose fail to maintain connectivity with their existing Thread partition,
    they will detach and start the attach procedure to look for neighboring devices.
  . A child detaches when it is not able to communicate with its parent.

I want to know the detailed condition for the detached role, and the source code location for that.
I'm using an OpenThread v1.1.

Best Regards,
Eric.

**Comment by jwhui** [comment, marked as answer]
The [Thread Specification](https://www.threadgroup.org/support#specifications) provides all of the details.

For relevant source code, take a look at [src/core/thread/mle.cpp](https://github.com/openthread/openthread/blob/main/src/core/thread/mle.cpp)
